### ISSUE TITLE
test cases/frameworks/1 boost broken

### ISSUE THREAD

**Issue by drsog** [mesonbuild/meson#2886]
Run on **MacOS** with --backend=**ninja**; with no Boost library installed.
Tested at master 0538009d , so unrelated to latest pull. 
Result: 
Code:
 
```
        #include <boost/version.hpp>
        #ifndef BOOST_LIB_VERSION
        # define BOOST_LIB_VERSION
        #endif
        "MESON_GET_DEFINE_DELIMITER"
BOOST_LIB_VERSION
Compiler stderr:
 /var/folders/80/d3v4src93cxc4h1v60gm521m0000gn/T/tmpb3__1wno/testfile.cpp:2:18: fatal error: 'boost/version.hpp' file not found
        #include <boost/version.hpp>
                 ^~~~~~~~~~~~~~~~~~~

```

Please Label as MacOS

**Comment by drsog** [comment]
[meson-test-run.txt](https://github.com/mesonbuild/meson/files/1609270/meson-test-run.txt)
Attached test log file

**Comment by bredelings** [comment]
The test as written will fail on any OS without boost installed, and it seems to be enabled for every OS where the CI tests will pass.  Since the travis mac images have boost installed, this test passes there.  So, are we sure this is a bug?

**Comment by drsog** [comment]
No, no certainty as yet. I only intended on the heads-up.  The snippit I saw had a directive checking for boost and it failed and tried to load it anyway.    Hopefully I can get more useful info soon.

**Comment by NickeZ** [comment]
Maybe it would be generally useful to be able to skip individual tests with a command line argument to the python test script. This would allow developers to have their own CI setups without frameworks installed.

**Comment by nirbheek** [comment]
> Maybe it would be generally useful to be able to skip individual tests with a command line argument to the python test script

I actually had patches for this in my tree. Will dig them out, clean them up and put them in a PR.

**Comment by nirbheek** [comment]
The original issue is fixed since https://github.com/mesonbuild/meson/pull/2894

**Comment by bonzini** [comment]
Too generic to be actionable.
